core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -478,12 +498,18 @@ public static SuggestMode resolve(final String str) {
     /** An enum representing the valid sorting options */
     public enum SortBy implements Writeable<SortBy> {
         /** Sort should first be based on score, then document frequency and then the term itself. */
-        SCORE,
+        SCORE((byte) 0x0),
         /** Sort should first be based on document frequency, then score and then the term itself. */
-        FREQUENCY;
+        FREQUENCY((byte) 0x1);
 
         protected static SortBy PROTOTYPE = SCORE;
 
+        private byte id;
+
+        SortBy(byte id) {
```

Comment:
++, I don't understand why we would need the byte representation here? I would get rid of the byte representation and id field. I see this `id()` and `fromId()` is used in serializing `TermSuggestion`, we could just use ` SortBy.PROTOTYPE.readFrom(in)` & `sort.writeTo(out)` just like we do in `TermSuggestionBuilder`? Regarding bwc, I this it is ok as this will in the next major version?